tests/validation.rs

```diff
@@ -200,10 +200,7 @@ fn total_fee_cant_overflow() {
         .change_output(AssetId::default())
         .build();
 
-    let mut interpreter = Interpreter::with_memory_storage();
-    let result = interpreter.transact(transaction);
-    assert!(matches!(
-        result,
-        Err(InterpreterError::ValidationError(ValidationError::ArithmeticOverflow))
-    ));
+    Interpreter::with_memory_storage()
+        .transact(transaction)
+        .expect("gas factor won't allow overflow");
 }
```

Comment:
We should have a test to verify that we properly round-up during the initial deduction, and round down for the refund

Reply:
We may also want a test that uses non-default gas price factors, to verify that the runtime params are being respected.

Reply:
Added in `tests/gas_factor.rs`